scripts/midterm.py

```diff
@@ -0,0 +1,514 @@
+    """
+    Function determines if features of a df is categorical or continuous
+    :param dataframe: Pandas dataframe
+    :return: dictionary
+    """
+    feature_type_dict = {}
+    for col in dataframe.columns:
+        if str(dataframe[col].dtype).startswith("int"):
+            if 2 <= dataframe[col].unique().size < int(len(dataframe) * 0.05):
+                feature_type_dict[col] = "categorical"
+            else:
+                feature_type_dict[col] = "continuous"
+        elif str(dataframe[col].dtype).startswith("float"):
+            feature_type_dict[col] = "continuous"
+        else:
+            feature_type_dict[col] = "categorical"
+    return feature_type_dict
+
+
+def get_df_as_matrix(
+    df, list_pred1, list_pred2, category, attribute, midpoint1=False, midpoint2=False
+):
+    """
+    Pivots a df into a matrix. Migpoint true shows interval center
+    :param df: Pandas df
+    :param list_pred1: columns categories
+    :param list_pred2: columns categories
+    :param category: measure
+    :param attribute:
+    :param midpoint1:
+    :param midpoint2:
+    :return:
+    """
+    list1 = list_pred1
+    list2 = list_pred2
+    df_list = []
+    for outer_bin in list1:
+        temp_list = []
+        for inner_bin in list2:
+            bin_array = str(outer_bin) + "," + str(inner_bin)
+            val = df.loc[df[category] == bin_array, attribute]
+            if val.size == 0:
+                temp_list.append(0)
+            else:
+                temp_list.append(val.iloc[0])
+        df_list.append(temp_list)
+
+    if midpoint1:
+        list1 = []
+        for interval in list_pred1:
+            list1.append(interval.mid)
+
+    if midpoint2:
+        list2 = []
+        for interval in list_pred2:
+            list2.append(interval.mid)
+
+    return pd.DataFrame(df_list, columns=list2, index=list1)
+
+
+def get_bin_intervals(list_1):
+    """
+    function returns a list of intervals from list
+    :param list_1: list of stepped values
+    :return: list of intervals
+    """
+    temp_list = []
+    for val in range(len(list_1) - 1):
+        j = pd.Interval(round(list_1[val], 3), round(list_1[val + 1], 3))
+        temp_list.append(j)
+
+    return temp_list
+
+
+def create_sheet(temp_list, col_names, excel_instance, file_name):
+    """
+    function adds sheet to instance of xlsxwriter and returns a df
+    :param temp_list: list of lists to create into pd df
+    :param col_names: list of col names
+    :param excel_instance: instance of xlsxwriter
+    :param file_name: file name
+    :return:
+    """
+    correlation_df = pd.DataFrame(temp_list, columns=col_names)
+    correlation_df = correlation_df.sort_values(by="Correlation", ascending=False)
+    correlation_df.to_excel(excel_instance, sheet_name=file_name, index=False)
+    return correlation_df
+
+
+def main():
+    # get data
+    df = pd.read_csv("../datasets/Titanic.csv")
+    df = df.drop("Name", axis=1)
+
```

Comment:
You tested this code on one dataset, but the assignment wants us to be able to run this on multiple datasets. main should take as parameters a pandas dataframe, a list of predictors, and a string of the name of the response variable, but in this section of your code, you are only running it on one dataset rather than making it modular